x/protorev/keeper/hooks.go

```diff
@@ -0,0 +1,235 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cltypes "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
+	gammtypes "github.com/osmosis-labs/osmosis/v15/x/gamm/types"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+	"github.com/osmosis-labs/osmosis/v15/x/protorev/types"
+)
+
+type Hooks struct {
+	k Keeper
+}
+
+var (
+	_ gammtypes.GammHooks = Hooks{}
+)
+
+// Create new ProtoRev hooks.
+func (k Keeper) Hooks() Hooks { return Hooks{k} }
+
+// ----------------------------------------------------------------------------
+// GAMM HOOKS
+// ----------------------------------------------------------------------------
+
+// AfterCFMMPoolCreated hook checks and potentially stores the pool via the highest liquidity method.
+func (h Hooks) AfterCFMMPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64) {
```

Comment:
I'm external to context around cosmwasmpool (release time, usage, etc.), but hooks would be great so that ProtoRev can backrun swaps from these pools.

Maybe can take the convo to slack / a call, but also would be interested if they interface with poolmanager / we can use them throughout the entire protorev system.

Reply:
Cosmwasm pools are in v16 scope but let's cut our protorev out of scope for cw pools.

We don't have a specific date set for the launch of the first cw pool anyways so we can add this if needed in v17. Lmk if any concerns on the suggestion 

The x/cosmwasmpool is plugged into the x/poolmanager similar to gamm and CL modules

Reply:
Got it, I'll add a note on my end to track it for v17.